plugins/inputs/slurm/slurm_test.go

```diff
@@ -0,0 +1,399 @@
+package slurm
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"testing"
+
+	"github.com/influxdata/telegraf/testutil"
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+	"github.com/stretchr/testify/require"
+)
+
+func TestURLs(t *testing.T) {
+	for _, url := range []string{"http://example.com:6820", "https://example.com:6820", "http://example.com"} {
+		plugin := Slurm{
+			URL: url,
+		}
+		require.NoError(t, plugin.Init())
+	}
+
+	for _, url := range []string{
+		"httpp://example.com:6820", "httpss://example.com:6820", "", "http://:6820", "http://",
+	} {
+		plugin := Slurm{
+			URL: url,
+		}
+		require.Error(t, plugin.Init())
+	}
+}
```

Comment:
Please split the two into one test for expected success and one test for expected failure. Furthermore, encapsulate the actual test (the inner of the `for` loop) with a `t.Run()` statement (like [here](https://github.com/influxdata/telegraf/blob/ad55f42d5366eb57e46e9232a637a84e8c23e9a5/plugins/inputs/http/http_test.go#L302)) to make sure the tests are run independently.

Reply:
Great! I added these changes on a5c910fa9171f381d048b267e3190c30bdbd0a09.